homeassistant/components/hydrawise/valve.py

```diff
@@ -0,0 +1,70 @@
+"""Support for Hydrawise sprinkler valves."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pydrawise.schema import Zone
+
+from homeassistant.components.valve import (
+    ValveDeviceClass,
+    ValveEntity,
+    ValveEntityDescription,
+    ValveEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import HydrawiseDataUpdateCoordinator
+from .entity import HydrawiseEntity
+
+VALVE_TYPES: tuple[ValveEntityDescription, ...] = (
+    ValveEntityDescription(
+        key="zone",
+        device_class=ValveDeviceClass.WATER,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Hydrawise valve platform."""
+    coordinator: HydrawiseDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        HydrawiseValve(coordinator, description, controller, zone)
+        for controller in coordinator.data.controllers.values()
+        for zone in controller.zones
+        for description in VALVE_TYPES
+    )
+
+
+class HydrawiseValve(HydrawiseEntity, ValveEntity):
+    """A Hydrawise valve."""
+
+    _attr_name = None
+    _attr_reports_position = False
+    _attr_supported_features = ValveEntityFeature.OPEN | ValveEntityFeature.CLOSE
+    zone: Zone
+
+    async def async_open_valve(self, **kwargs: Any) -> None:
+        """Open the valve."""
+        await self.coordinator.api.start_zone(self.zone)
+        self._attr_is_closed = False
```

Comment:
This statement makes the valve appear open.  Is there a way to wait for feedback from the controller, and have the state reflect what is reported from the controller?  That way if the command succeed but the valve is still closed, it won't be falsely reported.

Reply:
I would expect the API to return an error if the valve couldn't be opened, which would mean you wouldn't get to line 59.

I can remove that, but it also makes the tests a bit more annoying since you then have to wait on a data refresh to check that the state was updated. (which I guess is fine since the side-effect of validating the API was called is the important part?)

Reply:
Done